substrate-archive/src/actors/workers/blocks.rs

```
@@ -74,6 +74,29 @@ where
 		}
 	}
 
+	async fn fetch_genesis_block(&self) -> Result<Option<Block<B>>>
+	where
+		NumberFor<B>: From<u32>,
+	{
+		let backend = self.backend.clone();
+		if let Some(block) = backend.block(&BlockId::Number(0.into())) {
+			let cache = self.rt_cache.clone();
+			let spec = cache.get(block.block.hash())?.map(|ver| ver.spec_version).unwrap_or_default();
+			Ok(Some(Block { inner: block, spec }))
+		} else {
+			Ok(None)
+		}
+	}
+
+	async fn fetch_and_send_genesis_block(&self) -> Result<()> {
+		if let Some(genesis_block) = self.fetch_genesis_block().await? {
+			self.meta.send(genesis_block).await?;
+		} else {
+			log::error!("Couldn't fetch the genesis block from backend");
```

dq: why isn't this a hard error? I.e. shouldn't we return an error here?